### DISCUSSION TITLE
Keep using quantity instead of "contract size"

### DISCUSSION THREAD

**Discussion by klochowicz** [get10101/itchysats#1458] [General]
I think we all agree that our understanding of "quantity" has somewhat changed/evolved since the start of the project.
Recently we started renaming it to "n_contracts", "contracts", "contract size" etc. I have a feeling all of these names are somewhat awkward and redundant - "quantity" is used even by bitmex with definition *"Order quantity in units of the instrument (i.e. contracts)."*

In other words, could we keep the name "quantity", but align with the bitmex definition?

**Comment by bonomat** [comment]
You've my support on this :)

**Comment by thomaseizinger** [comment]
Quantity and contracts seem to be equivalent. Contract _size_ is different though, right?

I am pretty much ambivalent on quantity vs contracts although I have a slight preference for the unit-less contracts terminology. IMO it highlights better that you are not trading dollars but something that is worth 1 dollar (per contract).

**Reply by klochowicz** [reply]
IMHO it's not different, and we're inventing different terminology without a good reason. See Bitmex FAQ:

"For example, a contract with Lot Size of 100 would allow order quantities of 100, 200, 500, or 1000 contracts but would reject quantities of 1, 5, 10, 35, 125, or 250 contracts"

https://www2.bitmex.com/app/faq

it seems that order contract size == order quantity.

PS "Lot Size" would also be a much better name than size per parcel

**Reply by thomaseizinger** [reply]
> PS "Lot Size" would also be a much better name than size per parcel

Happy to adopt that!

The way I understood it is "contract size" == value per contract == $1 for us and quantity == number of contracts. From reading the page you linked, that seems to be consistent in that the FAQ refers to the number of contracts as a unit-less value. They are talking about orders with a quantity of 100 or 200 contracts.

I think using the term quantity is completely fine. I am just pointing out that the "unit" of quantity is _contracts_ in BitMex, .e. you place an order for 400 contracts, not for $400.

**Reply by klochowicz** [reply]
we have renamed parcel to "lot": https://github.com/itchysats/itchysats/pull/1587

**Comment by klochowicz** [comment]
It seems that no-one objected to keeping the term "quantity". Locking the discussion and marking it as "archived".

**Reply by luckysori** [reply]
I missed this discussion (I actuallly thought it was only about "parcel" vs "lot").

I also prefer "contracts" over "quantity". See https://github.com/itchysats/itchysats/pull/1779/commits/5034331a26b6dd41cf1e87f14b2c630463badbc4#diff-fcdf49cf28b52ed4be73680c9515cf4aadc3af124733b2cd4f9756dee6617d5cR983. We can revert that naming change if we're no longer trending in that direction though.

**Reply by bonomat** [reply]
I also didn't focus on this. Let's keep it open and put it into one of our iteration to come to a conclusion.

**Reply by klochowicz** [reply]
It would be good to reach conclusion soon, as it was open for over 2 months! :D

**Reply by klochowicz** [reply]
> I missed this discussion (I actuallly thought it was only about "parcel" vs "lot").
> 
> I also prefer "contracts" over "quantity". See [5034331#diff-fcdf49cf28b52ed4be73680c9515cf4aadc3af124733b2cd4f9756dee6617d5cR983](https://github.com/itchysats/itchysats/commit/5034331a26b6dd41cf1e87f14b2c630463badbc4#diff-fcdf49cf28b52ed4be73680c9515cf4aadc3af124733b2cd4f9756dee6617d5cR983). We can revert that naming change if we're no longer trending in that direction though.

that is a rather unfortunate change that I have missed in review ;/

in short, the original argument was that "quantity" is the more common term. `Units` would be perhaps good too, as together you could talk about "quantity of Units". 

I must say I'm not a fan of `Contracts` at all, because CFDs are *Contracts* For Difference. Happy to hear other opinions, or even get out-voted, but let's reach a decision soon. :)
